sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/intervalExpressions.scala

```diff
@@ -137,10 +139,23 @@ abstract class IntervalNumOperation(
 }
 
 case class MultiplyInterval(interval: Expression, num: Expression)
-  extends IntervalNumOperation(interval, num, multiplyExact, "multiplyExact")
+  extends IntervalNumOperation(interval, num) {
+
+  override protected def operation(interval: CalendarInterval, num: Double): CalendarInterval = {
+    if (checkOverflow) multiplyExact(interval, num) else multiply(interval, num)
```

Comment:
we shouldn't check the ansi flag for each row.

Reply:
we should define `operation` as a val: `val operation: (CalendarInterval, Double) => CalendarInterval`

Reply:
ah, thanks